src/WebForms/Compilation/BatchParser.cs

```diff
@@ -84,6 +110,8 @@ protected void AddDependency(VirtualPath virtualPath)
 
     internal abstract string DefaultDirectiveName { get; }
 
+    internal BaseTemplateParser TemplateParser { get => _templateParser; set => _templateParser = value; }
```

Comment:
would this make sense as an abstract method `GetTemplateParser()`?

Reply:
I have changed a bit, but basically made an abstract and let child to override... take a look